src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -689,8 +689,8 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
         /// <returns>The <see cref="MethodSymbol"/> of the Dispose method if one is found, otherwise null.</returns>
         internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNode syntaxNode, bool hasAwait, DiagnosticBag diagnostics)
         {
-            Debug.Assert(!(expr is null));
-            Debug.Assert(!(expr.Type is null));
+            Debug.Assert(expr is object);
```

Comment:
Isn't `(expr is {})` "easier"? I thought that's going to be the idiomatic null checking going forward.

Reply:
`expr is {}` does not error when `expr` is a `struct` while `is object` will error. It's strictly more correct.

Reply:
Seems like it just warns `warning CS0183: The given expression is always of the provided ('object') type`

Reply:
makes sense. so I think `is not null` would be sufficiently unclever for widespread use when available, instead of is object.

Reply:
I agree.  I'm actually not a huge fan of `is object` primarily because, for me, it heavily evokes a type check, when in reality all i'm really trying to convey is the null check in a safe way (i.e. without operator overloads coming into play).  So `is not null` is my pref as well for that as it reads fine and is completely clear about intent.

Reply:
ever since `notnull` became a valid generic constraint I have been contemplating whether it should be some kind of built-in pattern. (e.g. `expr is notnull`)

Reply:
I was surprised too, but for other reason. the constraint was ought to be `: not null`, then it'd align with the pattern. at least that's what we have in F#.

Reply:
I agree the `is object` expression screamed "type check" in my brain for a while. Eventually I got over it though and now just read it as a non-null check. The big bonus is that it works in literally every version of C#. While I do prefer `is not null` the best case scenario for that is C# 9.0.

Reply:
> While I do prefer is not null the best case scenario for that is C# 9.0.

Oh agreed.  For 8.0, i'm ok with this pattern.  If/when we get `is not null` i would prefer that be used.  I intend to do the IDE features here to help move to these patterns as well :)